bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/ConfigUtils.java

```diff
@@ -0,0 +1,143 @@
+ */
+
+package io.ballerina.runtime.internal.configurable.providers;
+
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.types.UnionType;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.internal.TypeConverter;
+import io.ballerina.runtime.internal.configurable.VariableKey;
+import io.ballerina.runtime.internal.configurable.exceptions.ConfigException;
+import io.ballerina.runtime.internal.types.BFiniteType;
+import io.ballerina.runtime.internal.types.BUnionType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import static io.ballerina.identifier.Utils.decodeIdentifier;
+import static io.ballerina.runtime.internal.errors.ErrorCodes.CONFIG_INCOMPATIBLE_TYPE;
+import static io.ballerina.runtime.internal.errors.ErrorCodes.CONFIG_UNION_VALUE_AMBIGUOUS_TARGET;
+
+/*
+ * Util class that contain methods that are common for env vars, cli configuration.
+ */
+public class ConfigUtils {
+
+    private ConfigUtils() {
+    }
+
+    public static BString getFiniteValue(VariableKey key, BUnionType unionType, String value, String variable) {
+        BString stringVal = StringUtils.fromString(value);
+        List<Type> memberTypes = unionType.getMemberTypes();
+        for (Type type : memberTypes) {
+            if (((BFiniteType) type).valueSpace.contains(stringVal)) {
+                return stringVal;
+            }
+        }
+        throw new ConfigException(CONFIG_INCOMPATIBLE_TYPE, variable, key.variable,
+                decodeIdentifier(unionType.toString()), value);
+    }
+
+    public static Object getFiniteBalValue(String strValue, BFiniteType finiteType, VariableKey key, String arg) {
+        List<Object> matchingValues = new ArrayList<>();
+        for (Object value : finiteType.getValueSpace()) {
+            String stringValue = value.toString();
+            if (stringValue.equals(strValue)) {
+                matchingValues.add(value);
+            }
+        }
+        if (matchingValues.size() == 1) {
+            return matchingValues.get(0);
+        }
+        String typeName = decodeIdentifier(finiteType.toString());
+        if (matchingValues.isEmpty()) {
+            throw new ConfigException(CONFIG_INCOMPATIBLE_TYPE, arg, key.variable, typeName, strValue);
+        }
+        throw new ConfigException(CONFIG_UNION_VALUE_AMBIGUOUS_TARGET, arg, key.variable, typeName);
+    }
+
+    public static boolean containsSupportedMembers(BUnionType unionType) {
+        for (Type memberType : unionType.getMemberTypes()) {
+            if (!isCliSupported(TypeUtils.getImpliedType(memberType).getTag())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean isCliSupported(int tag) {
```

Comment:
since this method is common to both CLI and env variable shall we have common name? or we can simple remove and return tag <= TypeTags.BOOLEAN_TAG || TypeTags.isXMLTypeTag(tag) in containsSupportedMembers method?